### DISCUSSION TITLE
회원정보수정을 하게되면 유저 데이터의 구조에서 fullName이 Null값으로 덮어씌워지는 현상

### DISCUSSION THREAD

**Discussion by hjlee2778** [prgrms-fe-devcourse/NFE1_2_2_MUX#49] [C. 질의 응답]
지금 마지막 커밋 기준으로 회원정보수정 자체에 대한 기능은 구현이 완료가 되었는데,
테스트로 새롭게 회원가입을 하여 정보수정을 했더니 로그인에서 문제가 발생함. (이 테스트를 통과해야 PR을 올릴 수 있을 것 같음)

![화면 캡처 2024-09-29 005237](https://github.com/user-attachments/assets/b8280cab-56c8-4c5c-b8e6-60759da50720)
이 사진이 첫 번째 회원가입 했을 때 저장되는 구조인데, 프로필 이미지를 변경하고 난 뒤의 구조는

![화면 캡처 2024-09-29 005300](https://github.com/user-attachments/assets/28b4eab4-8510-43ae-9995-50eeffc965d2)
다음과 같이 username 값이 추가됨과 동시에 fullName이 null값으로 변하는 사태 발생. 

여기저기 뒤져보면서 예상된 결과가 뭔가 처음 회원가입 할 때 nickname을 입력하여 생성하는 과정에서
api에 그 구조가 없기 때문에 이 부분을 필드에 사용자 정의 JSON 값을 넣어서 처리하는 방법으로 사용하고 있어서 그런가? 라는 의문을 가지게 되었음.

아직 더 테스트 해보면서 코드 수정해보고 있긴 한데, 혹시 이부분에 대해서 조금 더 잘 아시는 분 계시면 도움을 받고 싶어 올립니다..!
이 문제는 꼭 해결하고 넘어가야 할 것 같아서 완료하는데 까지 일정이 하루 늦어질 것 같습니다. 내일까지는 반드시 해결해보겠습니다!

**Comment by ryan-g00** [comment]
이게 무슨 문제일까 고민하고 있는 동안 희주님이 답을 찾아서 PR 에 내용 공유해두셨군욬ㅋㅋㅋㅋ 고생하셨습니다.
이 충돌 내용은 이슈로 한번 남겨주시는게 좋을 것 같아요
